src/babyjubjub-ecdsa/utils.ts

```diff
@@ -0,0 +1,53 @@
+import { EdwardsPoint, bytesToBigInt } from "babyjubjub-ecdsa";
+import { v4 as uuidv4, parse as uuidParse } from "uuid";
+import { JubmojiData } from "./types";
+
+export const serializeJumbojiData = (data: JubmojiData): string => {
+  return JSON.stringify({
+    jubmojiIndex: data.jubmojiIndex,
+    sig: data.sig,
+    msgNonce: data.msgNonce,
+    msgRand: data.msgRand,
+    R: data.R.serialize(),
+    T: data.T.serialize(),
+    U: data.U.serialize(),
+    leaderboardProof: data.leaderboardProof,
+  });
+};
+
+export const deserializeJubmojiData = (data: string): JubmojiData => {
+  const parsed = JSON.parse(data);
+  return {
+    jubmojiIndex: parsed.jubmojiIndex,
+    sig: parsed.sig,
+    msgNonce: parsed.msgNonce,
+    msgRand: parsed.msgRand,
+    R: EdwardsPoint.deserialize(parsed.R),
+    T: EdwardsPoint.deserialize(parsed.T),
+    U: EdwardsPoint.deserialize(parsed.U),
+    leaderboardProof: parsed.leaderboardProof,
+  };
+};
+
```

Comment:
Can we also have a succinctSerialize function? Where we just store the signature and pubkey index, optimized for wallet backups (including base64 encoding for now, but maybe we can do even better compression / decompression easily)

Reply:
I plan on create a separate PR for recovery and all its associated functions - will implement this there.